doc/manual/src/release-notes/rl-2.24.md

```diff
@@ -0,0 +1,300 @@
+
+  These changes aim to replace the use of autotools and make with Meson for building various components of Nix. Additionally, each library is built in its own derivation, leveraging Meson's "subprojects" feature to allow a single development shell for building all libraries while also supporting separate builds. This approach aims to improve productivity and build modularity, compared to both make and a monolithic Meson-based derivation.
+
+  Special thanks to everyone who has contributed to the Meson port, particularly [**@p01arst0rm**](https://github.com/p01arst0rm) and [**@Qyriad**](https://github.com/Qyriad).
+
+  Authors: [**John Ericson (@Ericson2314)**](https://github.com/Ericson2314), [**Tom Bereknyei**](https://github.com/tomberek), [**Théophane Hufschmitt (@thufschmitt)**](https://github.com/thufschmitt), [**Valentin Gagarin (@fricklerhandwerk)**](https://github.com/fricklerhandwerk), [**Robert Hensing (@roberth)**](https://github.com/roberth)
+  Co-authors: [**@p01arst0rm**](https://github.com/p01arst0rm), [**@Qyriad**](https://github.com/Qyriad)
+
+- Eval cache: fix cache regressions [#10570](https://github.com/NixOS/nix/issues/10570) [#11086](https://github.com/NixOS/nix/pull/11086)
+
+  This update addresses two bugs in the evaluation cache system:
+
+  1. Regression in #10570: The evaluation cache was not being persisted in `nix develop` because `evalCaches` retained references to the caches and was never freed.
+  2. Nix could sometimes try to commit the evaluation cache SQLite transaction without there being an active transaction, resulting in non-error errors being printed.
+
+  These bug fixes ensure that the evaluation cache is correctly managed and errors are appropriately handled.
+
+  Author: [**Lexi Mattick (@kognise)**](https://github.com/kognise)
+
+- Introduce `libnixflake` [#9063](https://github.com/NixOS/nix/pull/9063)
+
+  A new library, `libnixflake`, has been introduced to better separate the Flakes layer within Nix. This change refactors the codebase to encapsulate Flakes-specific functionality within its own library.
+
+  See the commits in the pull request for detailed changes, with the only significant code modifications happening in the initial commit.
+
+  This change was alluded to in [RFC 134](https://github.com/nixos/rfcs/blob/master/rfcs/0134-nix-store-layer.md) and is a step towards a more modular and maintainable codebase.
+
+  Author: [**John Ericson (@Ericson2314)**](https://github.com/Ericson2314)
+
+- CL options `--arg-from-file` and `--arg-from-stdin` [#9913](https://github.com/NixOS/nix/pull/9913)
```

Comment:
Is CL supposed to be CLI?

Reply:
Already fixed that in 22ad0e653f9abff03de97fe928eaf13a99f7e8a1.